internal/errors.go

```diff
@@ -0,0 +1,5 @@
+package internal
+
+import "errors"
+
+var ErrInvalidRecord = errors.New("received invalid record")
```

Comment:
is this error never going to be returned to an external consumer? re-export it or have it defined somewhere that can be referenced externally preferably

Reply:
It isn't currently. We were using and checking against it internally and I thought it'd be more recognizable when reading code to have it as a member of the internal package then as an unexported variable somewhere